inc/class-core-sitemaps-taxonomies.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Taxonomies sitemap.
+ *
+ * @package Core_Sitemaps
+ */
+
+/**
+ * Class Core_Sitemaps_Taxonomies.
+ * Builds the sitemap pages for Taxonomies.
+ */
+class Core_Sitemaps_Taxonomies extends Core_Sitemaps_Provider {
+	/**
+	 * Core_Sitemaps_Taxonomies constructor.
+	 */
+	public function __construct() {
+		$this->object_type = 'taxonomy';
+		$this->route       = '^sitemap-taxonomies-([A-z]+)-?([0-9]+)?\.xml$';
+		$this->slug        = 'taxonomies';
+	}
+
+	/**
+	 * Produce XML to output.
+	 */
+	public function render_sitemap() {
+		global $wp_query;
+
+		$sitemap  = get_query_var( 'sitemap' );
+		$sub_type = get_query_var( 'sub_type' );
+		$paged    = get_query_var( 'paged' );
+
+		$sub_types = $this->get_object_sub_types();
+
+		$this->sub_type = $sub_types[ $sub_type ]->name;
+		if ( empty( $paged ) ) {
+			$paged = 1;
+		}
+
+		if ( $this->slug === $sitemap ) {
+			if ( ! isset( $sub_types[ $sub_type ] ) ) {
+				// Invalid sub type.
+				$wp_query->set_404();
+				status_header( 404 );
+
+				return;
+			}
+
+			$url_list = $this->get_url_list( $paged );
+			$renderer = new Core_Sitemaps_Renderer();
+			$renderer->render_sitemap( $url_list );
+
+			exit;
+		}
+	}
+
+	/**
+	 * Get a URL list for a taxonomy sitemap.
+	 *
+	 * @param int $page_num Page of results.
+	 *
+	 * @return array $url_list List of URLs for a sitemap.
+	 */
+	public function get_url_list( $page_num ) {
+		$type = $this->sub_type; // Find the query_var for sub_type.
+		if ( empty( $type ) ) {
+			return;
+		}
+
+		// Get all of the taxonomies that are registered.
+		$taxonomies = $this->get_object_sub_types();
+
+		$url_list = array();
+
+		foreach ( $taxonomies as $taxonomy ) {
```

Comment:
I thought I'd already mentioned this, but I must have forgotten to add the comment. 

I don't think this loop is necessary in this method. I'd prefer to assume that validation for the sub-type is handled in the `render_sitemaps()` method so you can run the term query directly using `$this->sub_type` as the `'taxonomy'` argument.